### DISCUSSION TITLE
Matching Gitlab 13 Application Logs

### DISCUSSION THREAD

**Discussion by brazorfajeje** [wazuh/wazuh#14349] [Q&A] [answered]
Hi everyone,
I'm trying to create a custom decoder to parse gitlab 13 application.log, but I'm not sure to fully understand what is happening.

this is an example of a log related to the creation of a user:
```
2014-10-06T11:56:00.000Z: User "Administrator" (admin@example.com) was created
```

And this one of my tests:
```
<decoder name="gitlab-13-application-log">
  <prematch>\d\d\d\d-\d\d-\w+:\d\d:\d\d:\w+: </prematch>
</decoder>

<decoder name="gitlab-13-application-log-user">
  <parent>gitlab-13-application-log</parent>
  <prematch offset="after_parent">User "(\.+)" \((\.+)\)</prematch>
  <regex>(\d\d\d\d-\d\d-\w+:\d\d:\d\d:\w+): User "(\.+)" \((\.+)\) was created</regex>
  <order>timestamp, new_user, e-mail</order>
</decoder>
```

If I try testing the decoder I have:
```
**Phase 1: Completed pre-decoding. 
    full event:  2014-10-06T11:56:00.000Z: User "Administrator" (admin@example.com) was created  
    timestamp: 2014-10-06T11:56:00.000Z: User  
    hostname: - 
    program_name: - 

**Phase 2: Completed decoding. 
    name: - 
    data: "-" 
```

it seems that wazuh is already having difficulties in the pre-decoding phase, is it possible the problem is then not in my decoder? or in any case phase 2 should overwrite the timestamp field if the regex worked?

**Comment by xrisbarney** [comment, marked as answer]
Hi @brazorfajeje ,

Thank you for using Wazuh. 

I am testing the logs and decoders and will provide you with feedback shortly.

**Reply by xrisbarney** [reply]
Hi @brazorfajeje,

You are right that the problem is from the predecoding phase seeing this section `2014-10-06T11:56:00.000Z: User` as the timestamp.

This is due to the way the time is formatted from your gitlab and the way Wazuh static predecoders work. We are looking for a workaround.

However, Is it possible to change the way the timestamp is formatted for your GitLab logs/system?

**Reply by brazorfajeje** [reply]
I tried searching but couldn't find anything about it, I think it's not configurable. In general I think it might be interesting to support ISO 8601 and derivatives (8601.3, 8601.6 or 8601+timezone) format

**Reply by xrisbarney** [reply]
Thank you for the feedback. You can add more information about the ISO 8601 time format to this issue [here](https://github.com/wazuh/wazuh/issues/3525)
